enhancements/per-node-configuration.md

```diff
@@ -0,0 +1,484 @@
+# Per-Node Configuration with Node Selectors
+
+## Summary
+
+This enhancement proposes adding node selector support to OpenPERouter's `Underlay` resource, enabling different nodes to connect to different BGP neighbors and use node-specific network configurations. This allows OpenPERouter to support heterogeneous cluster topologies including multi-datacenter, multi-rack, and mixed-hardware environments.
+
+## Motivation
+
+Currently, OpenPERouter applies `Underlay` configuration cluster-wide, meaning all nodes in the cluster connect to the same set of BGP neighbors and use identical network interface configurations. This creates operational limitations in real-world deployments:
+
+- **Multi-rack deployments**: Different nodes are physically connected to different ToR (Top of Rack) switches but cannot be configured to peer with their respective ToRs
+- **Multi-datacenter clusters**: Nodes distributed across availability zones or datacenters need to peer with location-specific BGP routers
+- **Hardware heterogeneity**: Different server models have different NIC naming conventions (e.g., Dell uses `eno1`, HP uses `em1`), preventing a one-size-fits-all configuration
+- **Topology flexibility**: Cannot accommodate nodes in different physical network topologies within the same cluster
+
+### Goals
+
+- Enable per-node BGP neighbor configuration using Kubernetes node selectors
+- Support heterogeneous NIC configurations across different hardware platforms
+- Allow multiple `Underlay` resources to coexist, each targeting specific node subsets
+- Maintain backward compatibility with existing cluster-wide configurations
+- Prevent configuration conflicts through validation
+
+### Non-Goals
+
+- Per-neighbor node selectors (Option 2 - less flexible, documented as alternative)
```

Comment:
It might be me and my thick head, but I am finding it very easy to mix this non-goal with the goal above of "Enable per-node BGP neighbor configuration". Could you disambiguate these somehow ? Maybe just adjust the wording.

FWIW, it seems that you're specifically listing one thing that is proposed in one alternative as a non-goal - which AFAIU automatically discards said alternative. 

Just trying to figure out the scope here.

Reply:
> It might be me and my thick head, but I am finding it very easy to mix this non-goal with the goal above of "Enable per-node BGP neighbor configuration". Could you disambiguate these somehow ? Maybe just adjust the wording.
> 
> FWIW, it seems that you're specifically listing one thing that is proposed in one alternative as a non-goal - which AFAIU automatically discards said alternative. 
> 
> Just trying to figure out the scope here.

I will rephrase the non goals